lib/puppet/type/acl.rb

```diff
@@ -184,7 +195,7 @@ def insync?(is)
         s = p.tr '-', ''
         r << (s.sub!('r', '')?'r':'-')
         r << (s.sub!('w', '')?'w':'-')
-        r << (s.sub!('x', '')?'x':'-')
+        r << (s.sub!('x', '')?'x':'-')#'
```

Comment:
Things that can't be parsed unambiguously by either the ruby interpreter or by various syntax highlighters:
```ruby
(false|true?'a':'b')
```
`SyntaxError: unexpected tSTRING_BEG, expecting keyword_do or '{' or '('
(false|true?'a':'b')`

```ruby
(false|true ?'a':'b')
```
Works in the interpreter, but at least `pry` gets it wrong syntax highlighting (github too! Look at the closing paren)
```ruby
((false|true)?'a':'b')
```
Interpreter parses correctly as expected, but `?'a':` still seems to confuse some highlighters.

*sigh*